ui/raidboss/data/06-ew/trial/endsinger.ts

```diff
@@ -0,0 +1,251 @@
+import { TriggerSet } from '../../../../../types/trigger';
+
+// @TODO:
+// Interstellar - Position here can be gotten quicker via getCombatants on the chat line 'Tis so lonely between the stars'
+// current trigger doesn't give much time at all to dodge
+// Add phase triggers
+// Final phase triggers
+
+export interface Data extends RaidbossData {
+  storedStar?: { x: number; y: number };
+}
+
+const triggerSet: TriggerSet<Data> = {
+  zoneId: ZoneId.TheFinalDay,
+  timelineFile: 'endsinger.txt',
+  triggers: [
+    {
+      id: 'Endsinger Doomed Stars AoE',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: ['662E', '6634'], source: 'Doomed Stars', capture: true }),
+      alertText: (data, matches, output) => {
+        const x = parseFloat(matches.x);
+        const y = parseFloat(matches.y);
+
+        data.storedStar = { x, y };
+
+        if (x < 100) {
+          if (y < 100)
+            return output.se?.();
+
+          return output.ne?.();
+        }
+        if (y < 100)
+          return output.sw?.();
+
+        return output.nw?.();
+      },
+      outputStrings: {
+        ne: Outputs.northeast,
+        nw: Outputs.northwest,
+        se: Outputs.southeast,
+        sw: Outputs.southwest,
+      },
+    },
+    {
+      id: 'Endsinger Elegeia',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: ['662C', '6682'], source: 'The Endsinger', capture: false }),
+      response: Responses.aoe(),
+    },
+    {
+      id: 'Endsinger Fatalism',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '6632', source: 'The Endsinger', capture: false }),
+      condition: (data) => data.storedStar?.x !== undefined,
+      delaySeconds: 15,
+      alertText: (data, _matches, output) => {
+        const x = data.storedStar?.x;
+        const y = data.storedStar?.y;
+
+        delete data.storedStar;
+        if (!x || !y)
+          return undefined;
+
+        if (x < 100) {
+          if (y < 100)
+            return output.se?.();
+
+          return output.ne?.();
+        }
+        if (y < 100)
+          return output.sw?.();
+
+        return output.nw?.();
+      },
+      outputStrings: {
+        ne: Outputs.northeast,
+        nw: Outputs.northwest,
+        se: Outputs.southeast,
+        sw: Outputs.southwest,
+      },
+    },
+    {
+      id: 'Endsinger Galaxias',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '6C69', source: 'The Endsinger', capture: false }),
+      response: Responses.knockback(),
+    },
+    {
+      id: 'Endsinger Elenchos Middle',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '6644', source: 'The Endsinger', capture: false }),
+      response: Responses.getOut(),
+    },
+    {
+      id: 'Endsinger Elenchos Outsides',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '6642', source: 'The Endsinger', capture: false }),
+      response: Responses.getIn(),
```

Comment:
This is my own personal preferences, but "middle" vs "sides" is for stuff like this, and "in" vs "out" is for donut/circle mechanics.

```suggestion
      response: Responses.getMiddle(),
```

Reply:
I think I started with sides, and tried `getSides` and it didn't exist, and didn't think to check for `goSides`.